stsynphot/spparser.py

```diff
@@ -283,6 +284,16 @@ def p_arglist(self, tree):
         except AttributeError:
             pass  # We only care about this for relatively simple constructs.
 
+    @staticmethod
+    def _get_names_from_tree_values(args):
+        names = []
+        for arg in args:
+            try:
+                names.append(arg.meta['expr'])
+            except Exception:

```

Comment:
Wouldn't be safer to look for specific types of `Exception` instead of the generic base type? In the way it's here, the caller wouldn't ever be able to see errors that may happen inside the try-except block. One case that we can easily spot is a `KeyError` from the `meta['expr']` construct.